### DISCUSSION TITLE
Can I use 2.0 in it's current state for prototyping?

### DISCUSSION THREAD

**Discussion by M7mdKady14** [dialogic-godot/dialogic#1002] [Q&A] [answered]
I want to start writing the dialogue in my project and decided to choose dialogic and I tried the 1.4.3 but it just didn't have what I was looking for or was very hard and tedious to achieve.
then I looked into the new 2.0 and tried it on godot 3.5 and I have to say, I am mega impressed.
It has all things I wanted and more but can I use it in it's current state?
I don't mind bugs here and there since I don't plan to finish my game for at least a few months.

also it seems like some events are missing from the timeline editor like the end if branch and others, and If those are added automaticaly they aren't in my case

**Comment by coppolaemilio** [comment, marked as answer]
You will encounter many bugs and things are yet to be finalize. This means that some events could be renamed, syntax change or other stuff like that might break.
If you want to start working with 2.0 you could, but make sure to make a lot of commits, and whenever you are updating from version to version, you will likely have to go in and edit those files manually.
I would wait for the beta since from then on we will mostly be focusing on fixing bugs and the process of development will be easier for you, but if you want to give it a go now just beware that you might lose some data in the process.

**Reply by M7mdKady14** [reply]
That is fine by me.
also can you please give me a possible fix for the no events problem?

**Reply by coppolaemilio** [reply]
Not sure what events you are missing, but you have the `Condition` event which is an `if` statement and from it you can spawn the `elif` or `else` events as well.

**Reply by M7mdKady14** [reply]
But how?
The end of condition "event" doesn't show up when I make a condition event and I can't add it manually?
Am I missing something?

**Reply by coppolaemilio** [reply]
The visual style is not implemented yet, but this is kind of how it will look like. There is no "end of condition" event any more. As of now you will see just a boring white line, but that is the "end" of the condition

Mockup:
![Screen Shot 2022-07-13 at 14 26 00](https://user-images.githubusercontent.com/2206700/178733064-9f039792-2d06-44b3-ae61-fdd89a4f2d6c.png)

How it looks right now:
![image](https://user-images.githubusercontent.com/2206700/178733444-c0b8453a-a40c-434d-94a9-f8eb6e9170a1.png)

**Reply by M7mdKady14** [reply]
Ok but that's not how it looks on my end
![image](https://user-images.githubusercontent.com/104588917/178734542-2533988f-d6f3-4f0a-a03b-b44de24e978f.png)
there is no white line and when I try to add a second choice, it adds it under the first one (indented I mean)
I use godot 3.5.rc6
and this is a blank project with just a timeline

**Reply by coppolaemilio** [reply]
You are using the choice event, not the `condition` event

**Reply by M7mdKady14** [reply]
It's the same thing either way
![image](https://user-images.githubusercontent.com/104588917/178738984-da4015a5-b99d-4b65-b052-3b603f6acf09.png)

**Reply by coppolaemilio** [reply]
Make sure you have the latest version from 2.0 and if you do try starting the project from scratch

**Reply by M7mdKady14** [reply]
I just got the latest version and started another blank project but still no luck

**Reply by coppolaemilio** [reply]
are you using a custom editor theme? what rendering scale are you using?

**Reply by M7mdKady14** [reply]
I was using a custom editor theme (although it was just me changing the colors a bit and not an external download) but I tried using the normal one, reloaded my project and tried again. still no luck.
as for the rendering scale I don't know what you quite mean but I am using GLES3

**Reply by M7mdKady14** [reply]
ah apparently it now works??
will try different stuff to figure out what causes it

Edit: it seems it's something to do with godot editor themes, that or reloading the project but I think I can rule that out since I tried that before with no luck
it also seems random? I mean I tried a lot of stuff but sometimes it shows sometimes not? and sometimes it shows what is already there and when I create new condition events it doesn't show them.
also another thing I noticed it that I can't drag events into the space where the white line is supposed to be or at least close to it (even though it's not there)

**Reply by coppolaemilio** [reply]
Thanks! would love to know what is causing the issue so we can fix it

**Reply by M7mdKady14** [reply]
Critical information: I think it only works when i click the choice or condition event on the right and not when I drag and drop the event

https://user-images.githubusercontent.com/104588917/178756771-15b9f5e4-bc7c-4d0a-ae3f-53b30f8b0983.mp4

I think the source of the bug is from the drag and drop or something close to it.
hope this helps you solve the bug and thank you and your team for the wonderful addon

**Reply by coppolaemilio** [reply]
yes! that seems to be it, thanks for the report!
